src/api_server/swagger/firecracker.yaml

```diff
@@ -1018,6 +1022,10 @@ definitions:
         type: boolean
         description:
           When set to true, the vm is also resumed if the snapshot load is successful.
+      skip_compatibility_check:
+        type: boolean
+        description:
+          Optional field to skip the snapshot compatibility checks, which are on by default.
```

Comment:
Why do we need this knob? It adds a lot of complexity on the user/usage side as well Firecracker code complexity for handling all new cases introduced by it.

If the only production-worthy reason for it is to allow loading old snapshots, I suggest we look at some other approaches for migrating/upgrading those snaps, instead of adding and maintaining an extra knob in perpetuity..

Reply:
We could offer a dedicated snapshot compatibility tool for example, or simply warn when coming from TSC-unaware snaps but do best-effort and load them.

Reply:
I agree with Adrian, we should take a simpler approach to this issue, without adding a new parameter into the account. I also believe that validating v0.25 snapshots for tsc_freq should be done by default, while loading an older-version snapshot should just warn of tsc_freq not available. If my understanding is correct, vcpu_state.tsc_freq is automatically set to None if an older version snapshot is loaded.